test/badge.js

```diff
@@ -674,3 +675,225 @@ export async function deleteBadgesOverApi(n) {
     for(let i = 0; i < n; i++)
         await deleteBadgeOverApi();
 }
+
+/**
+ * Downloads the JSON file of a badge from the awarded badge details page.
+ * @returns The content of the json file as string
+ * @param {import('selenium-webdriver').ThenableWebDriver} driver
+ */
+const downloadBadgeJson = async (driver, badgeName) => {
+  const overflowMenu = await driver.findElement(
+    By.css('button:has(svg[icon="icon_more"])')
+  );
+  await overflowMenu.click();
+
+  const downloadButton = await driver.findElement(
+    ExtendedBy.tagWithText('button', `Download JSON-Datei (${currentBadgeStandardVersion})`)
+  );
+  await downloadButton.click();
+
+  await waitForDownload(driver, new RegExp(/^\d{4}-\d{2}-\d{2}-[a-zA-Z0-9_ ]+\.json$/));
+
+  const now = new Date();
+  const badgeJsonName =
+    `${now.getFullYear()}-` +
+    `${('0' + (now.getMonth()+1)).slice(-2)}-` +
+    `${('0' + (now.getDate())).slice(-2)}-` +
+    `${badgeName.replace(' ', '_')}.json`;
+  const file = fs.readFileSync(`${downloadDirectory}/${badgeJsonName}`, { encoding: 'utf-8' });
+  return file;
+};
+
+/**
+ * Checks if the badge is conform with the currently implemented
+ * open badges version and downloads the badges JSON file
+ * to check its conformity.
+ * This assumes that the driver already navigated to the badge detail page
+ * of an awarded badge.
+ * @param {import('selenium-webdriver').ThenableWebDriver} driver
+ */
+export const validateBadgeVersion = async (driver) => {
+  const badgeStandardText = await driver.findElement(
+      ExtendedBy.tagWithText("dt", "Badge-Standard")
+    );
+  const badgeStandardVersion = await driver.findElement(
+    ExtendedBy.sibling(badgeStandardText, By.css("dd"))
+  );
+  const badgeStandardVersionText = await badgeStandardVersion.getText();
+  assert.equal(badgeStandardVersionText, currentBadgeStandardVersion);
+
+  const file = await downloadBadgeJson(driver, testBadgeTitle);
+  
+  const badgeAsJson = JSON.parse(file);
+  assert.equal(typeof badgeAsJson['@context'] === "string", false);
```

Comment:
Maybe it would be nicer to user [`assert.notStrictEqual`](https://www.w3schools.com/nodejs/met_assert_notstrictequal.asp) here?

Reply:
Yes, good option, haven't really utilized this one much! Thanks :)